airflow/providers/common/sql/transfers/sql.py

```diff
@@ -0,0 +1,248 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import re
+from contextlib import closing
+from typing import TYPE_CHECKING, Any, Iterable, Mapping, Sequence
+
+from airflow.exceptions import AirflowException
+from airflow.hooks.base import BaseHook
+from airflow.models import BaseOperator
+from airflow.providers.common.sql.hooks.sql import DbApiHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+_PROVIDERS_MATCHER = re.compile(r"airflow\.providers\.(.*)\.hooks.*")
+
+_MIN_SUPPORTED_PROVIDERS_VERSION = {
+    "amazon": "4.1.0",
+    "apache.drill": "2.1.0",
+    "apache.druid": "3.1.0",
+    "apache.hive": "3.1.0",
+    "apache.pinot": "3.1.0",
+    "databricks": "3.1.0",
+    "elasticsearch": "4.1.0",
+    "exasol": "3.1.0",
+    "google": "8.2.0",
+    "jdbc": "3.1.0",
+    "mssql": "3.1.0",
+    "mysql": "3.1.0",
+    "odbc": "3.1.0",
+    "oracle": "3.1.0",
+    "postgres": "5.1.0",
+    "presto": "3.1.0",
+    "qubole": "3.1.0",
+    "slack": "5.1.0",
+    "snowflake": "3.1.0",
+    "sqlite": "3.1.0",
+    "trino": "3.1.0",
+    "vertica": "3.1.0",
+}
+
+
+class SqlToSqlOperator(BaseOperator):
+    """
+    Copy sql output data from one base to another database table.
```

Comment:
If this is Sql to Sql what is slack in the list of supported providers?

Reply:
I already answer you that Sql to sql operator is based on DbHookApi. Ale Hooks which are listed in code are compatible with. I found some issues with MySql on support of cursorfetchmany() but the same problem could exists in current generic Sql Operators.

Reply:
Approved - but indeed - why `Slack` here ? It does not seem to have DBAPIHook?

Reply:
`"slack": "5.1.0",` ?

Reply:
https://github.com/apache/airflow/blob/main/airflow/providers/common/sql/operators/sql.py - you have answer here (I had copied that part)

Reply:
Also qubole doesn't provide DbApiHook. I will remove them both from my file

Reply:
slack and qubole removed

Reply:
Yep.qubole has been removed from maintenance last month indeed.